.github/workflows/spelling2.yml

```diff
@@ -3,20 +3,70 @@ name: Spell checking
 on:
   pull_request_target:
   push:
+    branches: ["**"]
+    tags-ignore: ["**"]
 
 jobs:
   spelling:
     name: Spell checking
+    permissions:
+      contents: read
+      pull-requests: read
```

Comment:
This changes how the workflow works in a key way. In the old model, the workflow had a write token while it was processing the pull request. Now, while it does the processing, it takes a read token.